### PR TITLE
Add syntax and description to @layer

### PR DESCRIPTION
**Pull request by dipikabh**
#### Summary

- As a follow up after updating the @import page in https://github.com/mdn/content/pull/15205, this PR updates the @layer page to add "Syntax" and Description" sections for consistency.
- Also reorganised the content for better content flow.

This PR…
-->
- [ ] Adds a new document
- [X] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/css/@layer/index.md [hunk 1]
```diff
@@ -12,17 +12,33 @@ browser-compat: css.at-rules.layer
...
+  - : Is the name of the cascade layer.
+- _rules_
+  - : Is the set of CSS rules in the cascade layer.
+- _url_
+  - : Is a {{CSSxRef("&lt;string&gt;")}}, a `<url>`, or a {{CSSxRef("url")}} function representing the location of the resource to import. The URL may be absolute or relative.
 
-The `@layer` at-rule is used to create a cascade layer in one of three ways. The first is as in the example above, creating a block at-rule with the CSS rules for that layer inside.
+## Description
+
+Rules within a cascade layer cascade together, giving more control over the cascade to web developers. Any styles not in a layer are gathered together and placed into an anonymous layer that comes after all the declared layers. This means that any styles declared outside of a layer will override styles declared in a layer.
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:37]:
```suggestion
Rules within a cascade layer cascade together, giving more control over the cascade to web developers. Any styles not in a layer are gathered together and placed into an anonymous layer that comes after all the declared layers. This means that any styles declared outside of a layer will override styles declared in a layer, regardless of specificity.
```

File: files/en-us/web/css/@layer/index.md [hunk 2]
```diff
@@ -36,31 +52,41 @@ The `@layer` at-rule is used to create a cascade layer in one of three ways. The
...
+@layer {
+  p {
+    margin-block: 1rem;
+  }
+}
+```
+
+This creates an _anonymous cascade layer_. This layer functions in the same way as named layers; however, rules cannot be assigned to it later.
+
+Another way to create a cascade layer is by using {{cssxref("@import")}}. In this case, the rules would be in the imported stylesheet.
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:85]: should we remind readers that imports have to come first in a stylesheet?

**Review by estelle — COMMENTED** [review]
just a couple of questions

**Comment by dipikabh** [comment]
> just a couple of questions

Thanks for the review, @estelle. I have addressed the comments.

File: files/en-us/web/css/@layer/index.md [hunk 3]
```diff
@@ -12,17 +12,33 @@ browser-compat: css.at-rules.layer
...
 
 ```css
-@layer utilities {
-  /* creates a named layer called utilities. */
-}
+@layer layer-name {rules};
+@layer layer-name;
+@layer layer-name, layer-name, layer-name;
+@layer {rules};
+@import url layer( layer-name );
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:24]: Should the @import be in the @layer page?
And, if it is, should it come first as @import is supposed to come before, or precede, any other styles.

File: files/en-us/web/css/@layer/index.md [hunk 4]
```diff
@@ -12,17 +12,33 @@ browser-compat: css.at-rules.layer
...
+  - : Is the name of the cascade layer.
+- _rules_
+  - : Is the set of CSS rules in the cascade layer.
+- _url_
+  - : Is a {{CSSxRef("&lt;string&gt;")}}, a `<url>`, or a {{CSSxRef("url")}} function representing the location of the resource to import. The URL may be absolute or relative.
 
-The `@layer` at-rule is used to create a cascade layer in one of three ways. The first is as in the example above, creating a block at-rule with the CSS rules for that layer inside.
+## Description
+
+Rules within a cascade layer cascade together, giving more control over the cascade to web developers. Any styles not in a layer are gathered together and placed into an anonymous layer that comes after all the declared layers. This means that any styles declared outside of a layer will override styles declared in a layer, regardless of specificity.
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:37]:
```suggestion
Rules within a cascade layer cascade together, giving more control over the cascade to web developers. Any styles not in a layer are gathered together and placed into a single anonymous layer that comes after all the declared layers, named and anonymous. This means that any styles declared outside of a layer will override styles declared in a layer, regardless of specificity.
```

Comment (estelle) [reply files/en-us/web/css/@layer/index.md:37]: want to make sure people understand there can be multiple anonymous layers.

File: files/en-us/web/css/@layer/index.md [hunk 5]
```diff
@@ -36,31 +52,41 @@ The `@layer` at-rule is used to create a cascade layer in one of three ways. The
...
-
-You can also create a named cascade layer without assigning any styles. This can be a single name:
+The second way is to create a named cascade layer without assigning any styles. This can be a single layer, as shown below:
 
 ```css
 @layer utilities;
 ```
 
-Or, multiple layers can be defined at once. For example:
+Or multiple layers can be defined at once, as shown below:
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:61]:
```suggestion
Multiple layers can be defined at once, as shown below:
```

File: files/en-us/web/css/@layer/index.md [hunk 6]
```diff
@@ -36,31 +52,41 @@ The `@layer` at-rule is used to create a cascade layer in one of three ways. The
...
+
+```css
+@layer {
+  p {
+    margin-block: 1rem;
+  }
+}
+```
+
+This creates an _anonymous cascade layer_. This layer functions in the same way as named layers; however, rules cannot be assigned to it later.
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:83]:
```suggestion
This creates an _anonymous cascade layer_. This layer functions in the same way as named layers; however, rules cannot be assigned to it later. The order of precedence for anonymous layers is the order in which layers are declared, named or not, and lower than styles declared outside of a layer.
```

File: files/en-us/web/css/@layer/index.md [hunk 7]
```diff
@@ -36,31 +52,41 @@ The `@layer` at-rule is used to create a cascade layer in one of three ways. The
...
+
+Another way to create a cascade layer is by using {{cssxref("@import")}}. In this case, the rules would be in the imported stylesheet. Remember that the `@import` at-rule must precede all other types of rules, except the @charset rules.
```
Comment (estelle) [inline files/en-us/web/css/@layer/index.md:85]: I think it's enough to keep this and remove import from the syntax box at the start of this document.

**Review by estelle — CHANGES_REQUESTED** [review]
Looks good. let's get rid of the @import in the ## Syntax.

File: files/en-us/web/css/@layer/index.md (hunk 3 above)
Comment (dipikabh) [reply files/en-us/web/css/@layer/index.md:24]: Removed @import from the Syntax

**Comment by dipikabh** [comment]
> Looks good. let's get rid of the @import in the ## Syntax.

Thanks @estelle. I've incorporated the suggested changes.

**Review by estelle — APPROVED** [review]

**Pull request merged**